pkg/distributor/distributor.go

```diff
@@ -217,6 +237,42 @@ func shardByAllLabels(userID string, labels []client.LabelAdapter) (uint32, erro
 	return h, nil
 }
 
+// Remove the replica label from a slice of LabelPairs if it exists.
+func removeReplicaLabel(replica []byte, labels *[]client.LabelPair) {
+	for i := 0; i < len(*labels); i++ {
+		pair := (*labels)[i]
+		if pair.Name.Equal(replica) {
+			*labels = append((*labels)[:i], (*labels)[i+1:]...)
+			return
+		}
+	}
+}
+
+// Returns a boolean that indicates whether or not we want to remove the replica label going forward,
+// and an error that indicates whether we want to accept samples based on the cluster/replica found in ts.
+// nil for the error means accept the sample.
+func (d *Distributor) checkSample(ctx context.Context, ts client.PreallocTimeseries) (bool, error) {
+	cluster, replica := findHALabels([]byte(d.cfg.TrackerConfig.ReplicaLabel), []byte(d.cfg.TrackerConfig.ClusterLabel), ts.Labels)
+
+	// If the sample doesn't have either HA label, accept it.
+	// At the moment we want to accept these samples by default.
+	if cluster == "" || replica == "" {
+		return false, nil
+	}
+
+	// At this point we know we have both HA labels, we should lookup
+	// the cluster/instance here to see if we want to accept this sample.
+	err := d.replicas.CheckReplica(ctx, cluster, replica)
+	// CheckReplica should only have returned an error if there was a real error talking to Consul, or if the replica labels don't match.
+	if err != nil { // Don't accept the sample.
+		level.Error(util.Logger).Log("error", err)
```

Comment:
Can you add `"msg", "rejecting push due to replication"`?

Reply:
This is also going to spam the hell out of the logs.  Can you include a check for the status code, and not log 202s?

Reply:
Is there any way to check the status code `202` other than `httpgrpc.HTTPResponseFromError`? We probably want to check this in `consul_client.go`, we log the error in `cas` if retry is false.

Reply:
No, no other way.  Thats fine too.